test/integration/targets/sts_assume_role/tasks/main.yml

```diff
@@ -33,13 +33,19 @@
         aws_access_key: "{{ aws_access_key }}"
         aws_secret_key: "{{ aws_secret_key }}"
         security_token: "{{ security_token }}"
-        name: "ansible-test-sts-role"
+        name: "ansible-test-sts-{{ resource_prefix }}"
```

Comment:
The role name needs to be static so it can be created outside the tests. It shouldn't conflict between tests since it should already be present (once our CI account is updated).

Reply:
I think this can be dynamically created during the test - the IAM restrictions on role policy management/creation should be sufficient to let this be realsies-CI. 

Restrictions applicable: 

- ARN restrictions on the resource name `arn:aws:iam:{{account_id}}:role/ansible-test-sts-*`
- Attach/DetachRolePolicy restrictions on both the ARN of the role *and* the policy being attached/detached
- No permissions to create new policies independently
- ARN restrictions allow terminator-script to clean up afterwards


I'm thinking a policy would look something like this:

```
{
    "Sid": "AllowStsAssumeRoleTests",
        "Effect": "Allow",
        "Action": [
            "iam:CreateRole",
            "sts:AssumeRole",
            "iam:DeleteRole",
            "iam:DetachRolePolicy",
            "iam:ListInstanceProfilesForRole",
            "iam:AttachRolePolicy",
            "iam:CreateInstanceProfile"
        ],
        "Resource": [
            "arn:aws:iam::{{ aws_account_id }}:role/ansible-test-sts-*",
            "arn:aws:iam::{{ aws_account_id }}:instance-profile/ansible-test-sts-*"
        ]
},
{
    "Sid": "AllowStsAssumeRolePolicyManagement",
        "Effect": "Allow",
        "Action": [
            "iam:DetachRolePolicy",
            "iam:AttachRolePolicy",
        ],
        "Resource": [
            "arn:aws:iam::{{ aws_account_id }}:role/ansible-test-sts-*",
            "arn:aws:iam::{{ aws_account_id }}:instance-profile/ansible-test-sts-*"
        ],
        "Conditional": {"ArnLike": {"iam:PolicyArn": "arn:aws:iam:::policy/IamReadOnlyPolicy"}}
},
```

Reply:
This also makes it easier to run the integration tests on a dev's own account before it reaches CI since they don't have to first determine the test prerequisites.